broker/cloud_run/ztf/classify_upsilon/main.py

```diff
@@ -0,0 +1,139 @@
+
+    if has_min_detections_in_any_band:
+        # classify and publish results
+        alert_lite_df = _create_lite_dataframe(alert_lite.dict["alert_lite"])
+        upsilon_dict = _classify_with_upsilon(alert_lite_df)
+        TOPIC.publish(
+            pittgoogle.Alert.from_dict(
+                {
+                    "alert_lite": alert_lite.dict,
+                    "upsilon": {
+                        "objectId": alert_lite.dict["alertIds"]["objectId"],
+                        "candid": alert_lite.dict["alertIds"]["sourceId"],
+                        **upsilon_dict,
+                    },
+                },
+                attributes={
+                    **alert_lite.attributes,
+                    "pg_upsilon_g_label": upsilon_dict["g_label"],
+                    "pg_upsilon_g_flag": upsilon_dict["g_flag"],
+                    "pg_upsilon_r_label": upsilon_dict["r_label"],
+                    "pg_upsilon_r_flag": upsilon_dict["r_flag"],
+                    "pg_upsilon_i_label": upsilon_dict["i_label"],
+                    "pg_upsilon_i_flag": upsilon_dict["i_flag"],
+                },
+                schema_name="default",
+            )
+        )
+
+    return "", HTTP_204
+
+
+def _classify_with_upsilon(alert_lite_df: pd.Dataframe) -> dict:
+    upsilon_dict = {}
+    bands = alert_lite_df["filter"].map(pittgoogle.utils.ztf_fid_names()).unique()
+    rf_model = upsilon.load_rf_model()  # load UPSILoN's classification model
+    for band in bands:
+        # ---Extract data
+        filter_diaSources = alert_lite_df[alert_lite_df["filter"] == band]
+        mag_gt_zero = filter_diaSources["mag"].to_numpy() > 0
+        # set output to None if data is absent or there are too few data points for this band
+        # limit recommended by UPSILoN
+        if filter_diaSources.empty or mag_gt_zero.sum() <= 80:
+            upsilon_dict[f"{band}_label"] = None
+            upsilon_dict[f"{band}_probability"] = None
+            upsilon_dict[f"{band}_flag"] = None
+            continue
+
+        # ---Extract features
+        date = filter_diaSources["jd"].to_numpy()[mag_gt_zero]
+        mag = filter_diaSources["mag"].to_numpy()[mag_gt_zero]
+        mag_err = filter_diaSources["magerr"].to_numpy()[mag_gt_zero]
+        e_features = upsilon.ExtractFeatures(date, mag, mag_err)
+        e_features.run()
+        features = e_features.get_features()
+
+        # ---Classify
+        label, probability, flag = upsilon.predict(rf_model, features)
+        upsilon_dict[f"{band}_label"] = label
+        upsilon_dict[f"{band}_probability"] = probability
+        upsilon_dict[f"{band}_flag"] = flag
+
+    return upsilon_dict
+
+
+def _create_lite_dataframe(alert_dict: dict) -> pd.DataFrame:
+    """Return a pandas DataFrame containing the source detections."""
+
+    # sources and previous sources are expected to have the same fields
+    sources_df = pd.DataFrame([alert_dict.get("source")] + (alert_dict.get("prvSources") or []))
```

Comment:
```suggestion
    sources_df = pd.DataFrame([alert_dict.get("source")] + (alert_dict.get("prv_candidates") or []))
```

Double check this using an alert that has both of these.